midp/gfx.js

```diff
@@ -112,8 +112,34 @@ var currentlyFocusedTextEditor;
         return 1;
     };
 
+    var hasNewFrame = true;
+    var ctxToExecuteOnNewFrame;
+    function gotNewFrame(timestamp) {
+        if (ctxToExecuteOnNewFrame) {
+            ctxToExecuteOnNewFrame.execute();
+            ctxToExecuteOnNewFrame = null;
```

Comment:
Can *execute* eventually call *refresh0*? If so, *refresh0* will set *ctxToExecuteOnNewFrame*, after which this code will set that to `null`.